lib/AST/DeclContext.cpp

```diff
@@ -92,7 +92,11 @@ GenericTypeParamType *DeclContext::getProtocolSelfType() const {
     const_cast<ProtocolDecl *>(proto)->createGenericParamsIfMissing();
   }
 
-  return getGenericParamsOfContext()->getParams().front()
+  auto *genericParams = getGenericParamsOfContext();
+  if (genericParams == nullptr)
+    return nullptr;
```

Comment:
When does this come up for a protocol extension?

Reply:
Invalid code, like when an extension is inside of another decl, we don't build the generic parameter list for it anymore.

Reply:
But we still bind the underlying type? Okay.